src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -2222,7 +2222,7 @@ skip:
       NOTIFY_REQUEST_GET_OBJECTS::request req;
       bool is_next = false;
       size_t count = 0;
-      const size_t count_limit = m_core.get_block_sync_size(m_core.get_current_blockchain_height());
+      m_bss = m_core.get_block_sync_size(m_core.get_current_blockchain_height(), max_average_of_blocksize_in_queue());
```

Comment:
As mentioned in the other review, `m_bss` probably needs to be `atomic` as there is no thread safety afaik.

Reply:
Thanks. Will be fixed in updated PR.